source/html_attributeparser.c

```diff
@@ -1215,45 +1215,69 @@ valuetype_t HTML_ParseValueTypeAttribute(char* value)
     return valuetype;
 }
 
-void HTML_ParseAttributeContent(char* html_data, char* attribute_name,
+void HTML_ParseAttributeContent(char* html_data, char* attribute_name, bool has_value,
                                         attribute_t* attributes, int* offset)
 {
     // First off - we need to store the value in a char*,
-    // so let's do that.
+    // so let's prep that.
+    char* attr_value = NULL;
+    char* real_value = NULL;
+
+    if (has_value == true) {
+        // Before we begin, there's potential need to scrub the
+        // data at this point: we're particularly looking for
+        // whitespace and the start identifier ('"").
+        bool whitespace = true;
+        while(whitespace) {
+            if (html_data[*offset] != ' ') {
+                whitespace = false;
+                break;
+            } else {
+                *offset += 1;
+            }
+        }
 
-    // Allocate memory for the char* that holds the value.
-    char* attr_value = malloc(sizeof(char)*MAX_ATTR_VAL_LEN);
+        // Now, if we're at a '"', skip that too.
+        if (html_data[*offset] == '"')
+            *offset += 1;
 
-    // Store the offset index for the value parsing.
-    int val_offset = 0;
+        // Allocate memory for the char* that holds the value.
+        attr_value = malloc(sizeof(char)*MAX_ATTR_VAL_LEN);
 
-    // Just like BeginParse, iterate through the document.
-    int i;
-    for (i = 0; i < MAX_ATTR_VAL_LEN; i++) {
-        // End of value, break off.
-        if (html_data[i + *offset] == '"')
-            break;
+        // Store the offset index for the value parsing.
+        int val_offset = 0;
 
-        // Otherwise, append to value contents.
-        attr_value[val_offset] = html_data[i + *offset];
-        attr_value[val_offset + 1] = '\0';
-        val_offset++;
-    }
+        // Just like BeginParse, iterate through the document.
+        int i = 0;
```

Comment:
this is old practice should be in the for loop (try running with clang)

Reply:
`i` is referenced outside of the loop, therefore should not be delegated to initiation inside of it.